S3TransferUtilitySample/src/com/amazonaws/demo/s3transferutility/UploadActivity.java

```diff
@@ -518,84 +501,44 @@ private void beginUploadInBackground(String filePath) {
-                } else if ("video".equals(type)) {
-                    uri = MediaStore.Video.Media.EXTERNAL_CONTENT_URI;
-                } else if ("audio".equals(type)) {
-                    uri = MediaStore.Audio.Media.EXTERNAL_CONTENT_URI;
-                }
-                selection = "_id=?";
-                selectionArgs = new String[] {
-                        split[1]
-                };
-            }
-        }
-        if ("content".equalsIgnoreCase(uri.getScheme())) {
-            String[] projection = {
-                    MediaStore.Images.Media.DATA
-            };
-            Cursor cursor = null;
-            try {
-                cursor = getContentResolver()
-                        .query(uri, projection, selection, selectionArgs, null);
-                int column_index = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA);
-                if (cursor.moveToFirst()) {
-                    return cursor.getString(column_index);
-                }
-            } catch (Exception e) {
+    private File readContentToFile(Uri uri) throws IOException {
+        final File file = new File(getCacheDir(), getDisplayName(uri));
+        try (
+            final InputStream in = getContentResolver().openInputStream(uri);
+            final OutputStream out = new FileOutputStream(file, false);
+        ) {
+            byte[] buffer = new byte[1024];
+            for (int len; (len = in.read(buffer)) != -1; ) {
+                out.write(buffer, 0, len);
             }
-        } else if ("file".equalsIgnoreCase(uri.getScheme())) {
-            return uri.getPath();
+            return file;
         }
-        return null;
-    }
-
-    /**
-     * @param uri The Uri to check.
-     * @return Whether the Uri authority is ExternalStorageProvider.
-     */
-    public static boolean isExternalStorageDocument(Uri uri) {
-        return "com.android.externalstorage.documents".equals(uri.getAuthority());
     }
 
     /**
-     * @param uri The Uri to check.
-     * @return Whether the Uri authority is DownloadsProvider.
+     * Returns the filename for the given Uri
+     * @param uri the Uri
+     * @return String representing the file name (DISPLAY_NAME)
      */
-    public static boolean isDownloadsDocument(Uri uri) {
-        return "com.android.providers.downloads.documents".equals(uri.getAuthority());
-    }
-
-    /**
-     * @param uri The Uri to check.
-     * @return Whether the Uri authority is MediaProvider.
-     */
-    public static boolean isMediaDocument(Uri uri) {
-        return "com.android.providers.media.documents".equals(uri.getAuthority());
+    private String getDisplayName(Uri uri) {
+        final String[] projection = { MediaStore.Images.Media.DISPLAY_NAME };
+        try (
+            Cursor cursor = getContentResolver().query(uri, projection, null, null, null);
+        ){
+            int column_index = cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DISPLAY_NAME);
```

Comment:
`preferSnakeCase` for identifiers

Reply:
also an artifact of the previous implementation, but agreed!